spec/oauth2_api_client_spec.rb

```diff
@@ -17,7 +17,7 @@ def initialize
 end
 
 RSpec.describe Oauth2ApiClient do
-  before do
+  let(:auth_request_stub) do
```

Comment:
```suggestion
  let!(:auth_request_stub) do
```

and remove the `before` then please